### PR TITLE
 #145 Implement flag to force fragments handler to succeed

### PR DESCRIPTION
**Pull request by Voycawojka**
## Description
https://github.com/Knotx/knotx-fragments/issues/145

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: scenarios/failed-knotx-fragment/tasks.conf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        address = knotx.knot.te.pebble
+        deliveryOptions {
+          sendTimeout = 3000
+        }
+      }
+    }
+  }
+
+  allowInvalidFragmentsHeader = "Allow-Invalid-Fragments"
+  allowInvalidFragmentsParam = "allowInvalidFragments"
```
Comment (marcinus) [inline scenarios/failed-knotx-fragment/tasks.conf:34]: How about reusing `debug` parameter?

Comment (Voycawojka) [reply scenarios/failed-knotx-fragment/tasks.conf:34]: They exist independently. You can enable debug mode but do not allow invalid fragments to succeed

File: src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    });
+  }
+
+  @Test
+  @DisplayName("Fragments handler succeeds for a failed fragment when a param is provided")
+  @KnotxApplyConfiguration({"conf/application.conf",
+      "scenarios/failed-knotx-fragment/mocks.conf",
+      "scenarios/failed-knotx-fragment/tasks.conf",
+      "scenarios/failed-knotx-fragment/pebble.conf" })
+  void requestFailedDataWithHeader(VertxTestContext testContext, Vertx vertx, @RandomPort Integer delayedServicePort, @RandomPort Integer globalServerPort) {
```
Comment (marcinus) [inline src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java:73]: Please test enabling fault tolerance by header as well

File: src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package io.knotx.stack.functional;
```
Comment (marcinus) [inline src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java:16]: Please add licence header

File: conf/application.conf [hunk 4]
```diff
@@ -0,0 +1,11 @@
+########### Modules to start ###########
```
Comment (tomaszmichalak) [inline conf/application.conf:1]: All those files are not in the `src/test/resources` folder :) Please remove them.

File: scenarios/failed-knotx-fragment/tasks.conf (hunk 1 above)
Comment (tomaszmichalak) [reply scenarios/failed-knotx-fragment/tasks.conf:34]: Agree, every consumer can specify when it is enabled. This option is global.

File: src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(KnotxExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class FailedKnotxFragmentScenarioTest {
```
Comment (tomaszmichalak) [inline src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java:37]: What do you think about moving those scenarios to unit tests in knotx-fragments. Here, in knotx-stack, we can add new or update the `KnotxFragmentsDebugDataWithHandlebarsTest` scenario to have a failed fragment. Please note that we can do this with additional file in `KnotxApplyConfiguration`.

`KnotxFragmentsDebugDataWithHandlebarsTest` -> `KnotxFragmentsExecutionLogsTest`

Comment (Voycawojka) [reply src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java:37]: If I understand correctly then I don't agree. This functionality is independent from `debug` switch so I don't think they need to be in the same test suite

Comment (tomaszmichalak) [reply src/functionalTest/java/io/knotx/stack/functional/FailedKnotxFragmentScenarioTest.java:37]: You are right :)

**Pull request merged**